packages/launch/src/launch/index.tsx

```diff
@@ -1,45 +1,48 @@
 /**
  * External dependencies
  */
-import React, { useCallback } from 'react';
+import React from 'react';
 import { __ } from '@wordpress/i18n';
 import { Modal } from '@wordpress/components';
 import { Icon, wordpress } from '@wordpress/icons';
 import classNames from 'classnames';
-import { useSelect } from '@wordpress/data';
+import { useDispatch, useSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import FocusedLaunch from '../focused-launch';
 import LaunchContext from '../context';
 import { LAUNCH_STORE } from '../stores';
+import { FOCUSED_LAUNCH_FLOW_ID } from '../constants';
 import './styles.scss';
 
 interface Props {
-	onClose: () => void;
 	siteId: number;
 	locale: string;
 	redirectTo?: ( url: string ) => void;
+	openCheckout: ( siteId: number | string, isEcommerce?: boolean ) => void;
 }
 
 const FocusedLaunchModal: React.FunctionComponent< Props > = ( {
-	onClose,
 	siteId,
 	locale,
 	redirectTo = ( url: string ) => {
 		// Won't work if trying to redirect the parent frame
 		window.location.href = url;
 	},
+	openCheckout,
 } ) => {
 	const isModalDismissible = useSelect( ( select ) => select( LAUNCH_STORE ).isModalDismissible() );
 	const isModalTitleVisible = useSelect( ( select ) =>
 		select( LAUNCH_STORE ).isModalTitleVisible()
 	);
 
-	const onModalRequestClose = useCallback( () => {
-		isModalDismissible && onClose();
-	}, [ isModalDismissible, onClose ] );
+	const { closeFocusedLaunch } = useDispatch( LAUNCH_STORE );
+
+	const onModalRequestClose = () => {
+		closeFocusedLaunch();
```

Comment:
Why was the `isModalDismissible` flag check removed?

Reply:
We're already passing the flag to `Modal` at line 58 below so `onModalRequestClose` can't be called if the flag is set to `false`.

Reply:
> We're already passing the flag to Modal at line 58 below so onModalRequestClose can't be called if the flag is set to false.
Unfortunately, the `isDismissible` seems to only hide the `X` button, but it won't prevent the modal from closing when pressing the `ESC` key — you should be able to reproduce this behaviour by pressing the `ESC` key when in the Success view.

It looks like this could be fixed in a cleaner way by passing more props to the `<Modal>`:

```
  [...]
  shouldCloseOnEsc={ isModalDismissible }
  shouldCloseOnClickOutside={ isModalDismissible }
```

At that point, the `onModalRequestClose()` wrapper function would become useless, and we could pass the `closeFocusedLaunch()` function directly to the `<Modal>` too:

```
  onRequestClose={ closeFocusedLaunch }
```

Reply:
Added those 2 props and removed `onModalRequestClose` in https://github.com/Automattic/wp-calypso/pull/47601/commits/2f1f213d9ab9424d3bf7e8f5a1ec13da2a93442d